purchases/src/main/kotlin/com/revenuecat/purchases/PurchasesState.kt

```diff
@@ -7,5 +7,7 @@ internal data class PurchasesState(
     val allowSharingPlayStoreAccount: Boolean? = null,
     val updatedPurchaserInfoListener: UpdatedPurchaserInfoListener? = null,
     val purchaseCallbacks: Map<String, MakePurchaseListener> = emptyMap(),
-    val lastSentPurchaserInfo: PurchaserInfo? = null
+    val lastSentPurchaserInfo: PurchaserInfo? = null,
+    val appInBackground: Boolean = true,
```

Comment:
is the foreground call guaranteed to arrive the first time? is there a risk that we might be assuming background when we're actually in the foreground?

Reply:
yes, it's guaranteed since it's attached to the activity lifecycle. `onStop` can't happen before `onStart`